sdk/identity/azure-identity/azure/identity/_credentials/linux_vscode_adapter.py

```diff
@@ -0,0 +1,63 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import os
+import json
+import ctypes as ct
+from .._constants import VSCODE_CREDENTIALS_SECTION
+
+
+def _c_str(string):
+    return ct.c_char_p(string.encode('utf-8'))
+
+
+try:
+    _libsecret = ct.cdll.LoadLibrary('libsecret-1.so.0')
+    _libsecret.secret_schema_new.argtypes = \
+        [ct.c_char_p, ct.c_uint, ct.c_char_p, ct.c_uint, ct.c_char_p, ct.c_uint, ct.c_void_p]
+    _libsecret.secret_password_lookup_sync.argtypes = \
+        [ct.c_void_p, ct.c_void_p, ct.c_void_p, ct.c_char_p, ct.c_char_p, ct.c_char_p, ct.c_char_p, ct.c_void_p]
+except OSError:
+    _libsecret = None
+
+
+def _get_user_settings_path():
+    app_data_folder = os.environ['HOME']
+    return os.path.join(app_data_folder, ".config", "Code", "User", "settings.json")
+
+
+def _get_user_settings():
+    path = _get_user_settings_path()
+    try:
+        with open(path) as file:
+            data = json.load(file)
+            environment_name = data.get("azure.cloud", "Azure")
+            return environment_name
+    except IOError:
+        return "Azure"
+
+
+def _get_refresh_token(service_name, account_name):
+    # _libsecret.secret_password_lookup_sync raises segment fault on Python 2.7
+    # temporarily disable it on 2.7
+    import sys
+    if sys.version_info[0] < 3:
+        return None
```

Comment:
If we go down this path we ultimately get the error message "No Azure user is logged in to Visual Studio Code". That's disinformation. If we support only 3.x for this feature, the error message must indicate that. First solution that comes to my mind (feel free to implement a better one) is to raise `NotImplementedError` here and handle it in `get_token`.